doc/docs-revamp/dev-onboarding-quick-setup.md

```diff
@@ -0,0 +1,375 @@
+
+1. Create a new directory for your project and navigate to it in your terminal.
+
+2. If you are using Nix, enter a Nix shell that provides the necessary development environment by running `nix develop` in your project directory. If you are not using Nix, make sure that all required C libraries are installed since PlutusTx depends on `cardano-base`, which in turn depends on cryptographic C libraries like `libblst`, `libsecp256k1`, and `libsodium`. 
+
+3. Create a new directory named `src` and a new file named `VestingContract.hs` inside it. This is where we'll write our smart contract code using Plutus Tx.
+
+## First smart contract example
+
+### Writing a simple Vesting contract
+
+For this example, we will walk through a smart contract that is designed to use only very limited conditions. The contract provides for someone to send a gift of ada to a beneficiary. The beneficiary can access their gift only after a specific deadline has been passed and the beneficiary has signed the contract. 
+
+### Overview 
+
+#### High-level conceptual stages of the vesting contract
+
+| Stage | Description |
+|-------|-------------|
+| A. Initialization | Define the data structure and data types for the vesting contract, including the beneficiary's public key hash and the deadline for the vesting period. |
+| B. Validation Logic | Create the logic that will validate transactions attempting to withdraw funds from the contract. |
+| C. Validation Conditions | Establish the conditions under which the funds can be released, specifically that the current time is beyond the deadline and that the transaction is signed by the beneficiary. |
+| D. Compilation | Wrap and compile the validation logic into a Plutus Validator type. |
+| E. Helper Functions | Provide additional functions to save the compiled validator to a file and to print the vesting datum in JSON format for easy inspection. |
+
+#### Summarized list of the vesting contract's functionality section by section
+
+1. Importing necessary modules and functions
+2. On-chain / Validator code
+   - Define the `VestingDatum` data type that holds the beneficiary's public key hash and the deadline for fund release.
+   - Implement the core `mkVestingValidator` function that checks if the transaction is signed by the beneficiary and if the deadline has passed.
+   - Wrap the validator function and compile it to a Validator type. `mkWrappedVestingValidator` is a wrapper for the validator function to conform to the expected type signature for Plutus validators. The compiled validator script ready for deployment on the blockchain.
+3. Helper functions
+   - `saveVal` is a helper function that saves and writes the validator script to a file. 
+   - `printVestingDatumJSON` is a helper function to print the vesting datum in JSON format, given a public key hash and a deadline in ISO 8601 format.
+
+### Section 1: Importing the required modules and functions
+
+```haskell
```

Comment:
We shouldn't have code inlined in the md files. The code needs to be put in a Haskell module and checked by CI.

Reply:
OK, good to know. Once we agree on the example we want to use here, would we use code block inclusions instead of inlining code to show each section we want to describe? I'm not sure how we would follow that practice while reviewing drafts. But once we have a documentation platform established for the revamp, then I imagine we could generate draft pages that include code blocks.